numpy/lib/index_tricks.py

```diff
@@ -195,12 +196,12 @@ def __getitem__(self, key):
             if start is None:
                 start = 0
             if isinstance(step, (_nx.complexfloating, complex)):
-                step = abs(step)
-                length = int(step)
+                # Prevent the (potential) creation of integer arrays
+                step = length = _nx.floor(abs(step))
```

Comment:
Why `floor` instead of `int`?

Reply:
If `step == 1` then this could lead to a situation where an integer array is returned (if `start` and `stop` are integers), while previously it would _always_ return a float array.
`floor()` ensures that `step` remains a float.
I suppose `float(int(abs(step)))` would also do the trick.

Reply:
@BvB93 do you intend to change this?

Reply:
Done as of https://github.com/numpy/numpy/pull/16971/commits/4dcc8408efdffc6181bc67cc72fe6741b3eeae1d.